### PR TITLE
{tool}[GCCcore-12.3.0] datalad-0.19.5

### PR DESCRIPTION
**Pull request by jkuhl-uni**
Hi, with this, I'm updating datalad for the GCCcore 12.3.0 toolchain, as per request in #19879

### CODE REVIEW

**Comment by jfgrimm** [comment]
@boegelbot: please test @ jsc-zen3

**Comment by jkuhl-uni** [comment]
Hey @jfgrimm,
Thanks for having a look at this :)
It seems tome as if the C++ ISO is too new, I trieed to fix that. could you try to run it again?

**Comment by jfgrimm** [comment]
@boegelbot please test @ jsc-zen3

**Comment by jfgrimm** [comment]
@boegelbot: please test @ jsc-zen3

**Comment by jfgrimm** [comment]
@boegelbot: please test @ generoso

**Review by jfgrimm — APPROVED** [review]
lgtm

File: easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+homepage = 'https://git-annex.branchable.com'
+description = """git-annex allows managing large files with git, without storing the file contents in git. It can sync,
+backup, and archive your data, offline and online. Checksums and encryption keep your data safe and secure. Bring the
+power and distributed nature of git to bear on your large files with git-annex."""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+dependencies = [
+    ('GHC', '9.4.6', '-x86_64', SYSTEM),
+    ('GCC', '12.3.0', '', SYSTEM),
```
Comment (jfgrimm) [inline easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: wait, why the runtime dependency on a `SYSTEM` compiled `GCC`, when we're using the `GCCcore` toolchain?

**Review by jfgrimm — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb (hunk 1 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]:
```suggestion
```

Comment (jkuhl-uni) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: Good question, thanks for noticing. That flew under my radar. You cannot do it without GCC (at least i get an error from Stak if there is no GCC loaded...), I just tried.
I put it instead in the GCC toolchain. I'm waiting for it to compile, I'll commit the changes once it worked on the cluster here.

Comment (jfgrimm) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: you might be able to do it with `GCCcore` if you add `binutils` as a build dependency

Comment (jkuhl-uni) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: Thanks for the suggestion. I tried, however unfortunately, there is still a problem with Stack 2.11.1 and GHC 9.4.6. I updated stack, but the ghc of stack gets stuck in the linker phase of GCC... are there any workarounds?

Comment (jfgrimm) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: removing `GCC` and adding `binutils` as a build dependency works for me

Comment (jfgrimm) [inline easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]:
```suggestion
builddependencies = [('binutils', '2.40')]

dependencies = [
    ('GHC', '9.4.6', '-x86_64', SYSTEM),
```

Comment (jfgrimm) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: to clarify, I'm happy for `GHC` to stay at the system compiler level, this is purely about the explicit `('GCC', '12.3.0', '', SYSTEM),` dependency in `git-annex`

Comment (jkuhl-uni) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: Sure, that makes sense, I agree.
I was already wondering why GHC is not in the GCCcore toolchain, but this was already discussed in #11310.

Comment (jkuhl-uni) [reply easybuild/easyconfigs/g/git-annex/git-annex-10.20230802-GCCcore-12.3.0.eb:15]: Yep, my fault. I updated Stack, anyway. I'll send over the changes

**Comment by jfgrimm** [comment]
@boegelbot: please test @ generoso

**Comment by jfgrimm** [comment]
@boegelbot: please test @ jsc-zen3

**Review by jfgrimm — APPROVED** [review]

**Comment by jfgrimm** [comment]
Going in, thanks @jkuhl-uni!

**Pull request merged**